lua/treesitter-context/render.lua

```diff
@@ -305,25 +335,11 @@ local function close(context_winid)
     end
 
     local bufnr = api.nvim_win_get_buf(context_winid)
-    if bufnr ~= nil then
-      table.insert(retired_buffers, bufnr)
-    end
-    if api.nvim_win_is_valid(context_winid) then
-      api.nvim_win_close(context_winid, true)
-    end
-
-    if fn.getcmdwintype() ~= '' then
-      -- Can't delete buffers when the command-line window is open.
-      return
-    end
-
-    -- Delete retired buffers.
-    for _, retired_bufnr in ipairs(retired_buffers) do
-      if api.nvim_buf_is_valid(retired_bufnr) then
-        api.nvim_buf_delete(retired_bufnr, { force = true })
-      end
+    api.nvim_win_close(context_winid, true)
+    if bufnr ~= nil and api.nvim_buf_is_valid(bufnr) then
+      table.insert(buffer_pool, bufnr)
```

Comment:
It seems that the only entry point to actually inserting a buffer into the pool is here, so can't we avoid need for the `delete_excess_buffers()` call by simply not inserting a buffer if the pool is at capacity?

Reply:
We can't delete buffers in-place, when the command-line window is open.

Reply:
That's not what is being suggested?

Reply:
As far as I understand, no. If the pool is at capacity and we don't add a buffer, we'll need to delete one. However, if the command-line window is open, we cannot proceed with the deletion. Regardless, sometimes we have to add the buffer to the pool even if the pool is full. While I could introduce another 'if' branch, I doubt it would enhance performance.

Reply:
However, I'm not certain it's a good idea to check all buffers in the pool within the delete_excess_buffers function. We might just delete the last buffer in the list instead. I don't believe the buffers will become invalid without a reason.